guides/source/active_support_core_extensions.md

```diff
@@ -837,6 +839,55 @@ In the previous example the macro generates `avatar_size` rather than `size`.
 
 NOTE: Defined in `active_support/core_ext/module/delegation.rb`
 
+#### `delegate_missing_to`
+
+If you need to decorate an object and delegate all of the method calls to it:
+
+```ruby
+class Partition
+  def initialize(event)
+    @event = event
+  end
+
+  def person
+    @event.detail.person || @event.creator
+  end
+
+  private
+    def respond_to_missing?(name, include_private = false)
+      @event.respond_to?(name, include_private)
+    end
+
+    def method_missing(method, *args, &block)
+      @event.send(method, *args, &block)
+    end
+end
+```
+
+The `delegate_missing_to` macro can help you condense the example to:
+
+```ruby
+class Partition
+  delegate_missing_to :@event
+
+  def initialize(event)
+    @event = event
+  end
+
+  def person
+    @event.detail.person || @event.creator
+  end
+end
+```
+
+The target can be anything callable within the object, e.g. instance
+variables, methods, constants, etc.
+
+WARNING: The delegated method calls must be public on the target, otherwise
+they will raise `NoMethodError`.
```

Comment:
This warning demonstrates that the behaviour doesn't match your manual implementation. :wink:

Really though, I think the full `method_missing`/`respond_to_missing?` illustration is too much unnecessary detail. It works for `delegate`, but isn't an efficient use of space in general. Let's ditch it, and just focus on usage.

Maybe also consider reusing the User/Profile example... "or you could just forward everything" seems a logical leap from "here's how to delegate 4 methods in one line"